internal/constants/scripts.go

```diff
@@ -66,140 +70,156 @@ function map_host_macs_to_interfaces() {
+  if [ "$(ls -A $host_work_dir )" ] ; then
+    for mac_address in $(cat $mapping_file | cut -d= -f1 | tr '[:lower:]' '[:upper:]'); do
+      host_nic="${host_macs_to_hw_iface[${mac_address}]:-}"
+      if [[  -n "$host_nic" ]]; then
+        logical_nic="${logical_mac_nic_map[${mac_address}]:-}"
+        for nmfile in $host_work_dir/*.nmconnection ; do
+          sed -i -e "s/=$logical_nic$/=$host_nic/g" -e "s/=$logical_nic\./=$host_nic\./g" $nmfile
+          echo "Info: Using logical interface name '$logical_nic' for interface with Mac address '$mac_address', updated $nmfile"
+        done
+      fi
+    done
+  fi
+}
 
-    mapping_file="${d}/${MAC_NIC_MAPPING_FILE}"
+function process_host_directories_by_mac_address() {
+  pattern="$(echo -n ${!host_macs_to_hw_iface[@]} | sed 's/ /|/g')"
+  if [[ -z "$pattern" ]] ; then
+    return
+  fi
+  src_dir=$(mktemp -d)
+  cp -r ${NMCONNECTIONS_DIR}/host* $src_dir || /bin/true
+  for host_src_dir in $(ls -1 -d $src_dir/host* || echo)
+  do
+    mapping_file="${host_src_dir}/${MAC_NIC_MAPPING_FILE}"
     if [[ ! -f "$mapping_file" ]]
     then
       echo "Warning: Mapping file $mapping_file is missing. Skipping on directory $d"
       continue
     fi
 
-    if [[ -z "$(ls -A $d/*.nmconnection)" ]]
+    if [[ -z "$(ls -A $host_src_dir/*.nmconnection)" ]]
     then
       echo "Warning: Host directory does not contain any nmconnection files, skipping"
       continue
     fi
 
-    # check if mapping file contains mac-address that exists on the current host
-    for mac_address in $(cat $mapping_file | cut -d= -f1 | tr '[:lower:]' '[:upper:]')
-    do
-      if [[ ! -z "${host_macs_to_hw_iface[${mac_address}]:-}" ]]
-      then
-        host_dir=$(mktemp -d)
-        cp ${d}/* $host_dir
-        echo "Info: Found host directory: $d, configuration copied to $host_dir"
-        return
-      fi
-      echo "Info: No match found in host macs for '$mac_address'"
-    done
-  done
+    if grep -q -i -E "$pattern" $mapping_file; then
 
-  if [ -z "$host_dir" ]
-  then
-    echo "Error: None of the host directories contained a mac-address to host mapping for the current host"
-    // We should not exit the script here in any fashion as the Dracut initqueue handler will not run any subsequent scripts if there is an exit.
-  fi
-}
+      host_work_dir=$(mktemp -d)
+      # check if mapping file contains mac-address that exists on the current host
+      copy_physical_nics_files $host_src_dir $mapping_file $host_work_dir
 
-function set_logical_nic_mac_mapping() {
-  echo "Info: Checking '$mapping_file' for logical nic to mac mappings"
-  # initialize logical_nic_mac_map with mapping file entries
-  readarray -t lines < "${mapping_file}"
-  for line in "${lines[@]}"
-  do
-    mac=${line%%=*}
-    nic=${line#*=}
-    logical_nic_mac_map[$nic]=${mac^^}
+      # Copy the rest of the files from the host directory if a file was copied from the host directory
+      copy_other_files $host_src_dir $host_work_dir
```

Comment:
I notice that a lot of the echo statements that were in the previous implementation have been removed in this PR. 

The original intent of these echo statements is to provide some useful output in the cases where there is failure. This was added deliberately.

I think we should make an effort to continue to give verbose feedback on what we are processing.